### PR TITLE
vertx package and SessionFactory.close()

### PR DESCRIPTION
**Pull request by gavinking**
This work:

- moves @Sanne's new code for managing the `Vertx` instance into a public package, and renames the classes/interfaces
- introduces `SessionFactory.close()`, to clean up vert.x and the `Pool`

Does this look good, @Sanne @DavideD?

### CODE REVIEW

File: example/src/main/java/org/hibernate/example/reactive/Main.java [hunk 1]
```diff
@@ -79,7 +79,7 @@ public static void main(String[] args) {
 		)
 				.toCompletableFuture().join();
 
-		System.exit(0);
+		sessionFactory.close();
```
Comment (Sanne) [inline example/src/main/java/org/hibernate/example/reactive/Main.java:82]: whoa, why did you have a Sytem.exit there? :)

Comment (gavinking) [reply example/src/main/java/org/hibernate/example/reactive/Main.java:82]: 'Cos I had no way to shut down Vert.x before :-)

File: hibernate-reactive-core/src/main/java/org/hibernate/reactive/vertx/VertxInstance.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package org.hibernate.reactive.vertx;
+
+import io.vertx.core.Vertx;
+import org.hibernate.service.Service;
+
+/**
+ * Used by {@link org.hibernate.reactive.pool.impl.SqlClientPool}
```
Comment (Sanne) [inline hibernate-reactive-core/src/main/java/org/hibernate/reactive/vertx/VertxInstance.java:7]: :100:

File: example/src/main/java/org/hibernate/example/reactive/Main.java (hunk 1 above)
Comment (Sanne) [reply example/src/main/java/org/hibernate/example/reactive/Main.java:82]: that's what I suspected.. so you already had a clue that something was leaking :)

Comment (gavinking) [reply example/src/main/java/org/hibernate/example/reactive/Main.java:82]: C'mon, all good Java programs call `System.exit()`, right @emmanuelbernard?

Comment (Sanne) [reply example/src/main/java/org/hibernate/example/reactive/Main.java:82]: now you're both in that club! :-)

Comment (emmanuelbernard) [reply example/src/main/java/org/hibernate/example/reactive/Main.java:82]: > C'mon, all good Java programs call `System.exit()`, right @emmanuelbernard?

It's an underrated API.

**Pull request merged**